inttest/openebs/openebs_test.go

```diff
@@ -0,0 +1,179 @@
+	common.BootlooseSuite
+}
+
+func (s *OpenEBSSuite) TestK0sGetsUp() {
+	ctx := s.Context()
+	s.T().Run("Both storage and helm extensions enabled", func(t *testing.T) {
+		s.PutFile(s.ControllerNode(0), "/tmp/k0s.yaml", k0sConfigWithBoth)
+		s.Require().NoError(s.InitController(0, "--config=/tmp/k0s.yaml", "--disable-components=konnectivity-server,metrics-server"))
+		s.Require().NoError(s.RunWorkers())
+
+		kc, err := s.KubeClient("controller0", "")
+		s.Require().NoError(err)
+
+		err = s.WaitForNodeReady("worker0", kc)
+		s.NoError(err)
+
+		err = s.WaitForNodeReady("worker1", kc)
+		s.NoError(err)
+
+		// When both storage and helm are enabled, there should be no action.
+		// Unfortunately to check that, the best we can do is seeing that it's not
+		// created after a grace period
+		s.T().Log("Waiting 30 additional seconds of grace period to see if charts are created")
+		time.Sleep(30 * time.Second)
+
+		s.T().Log("Checking that the chart isn't created")
+		hc, err := s.HelmClient("controller0", "")
+		s.Require().NoError(err)
+		_, err = hc.Charts("kube-system").Get(ctx, openEBSChart, metav1.GetOptions{})
+		s.Require().True(errors.IsNotFound(err), "Chart was created when it shouldn't have been")
+
+		s.T().Log("Stopping k0s")
+		s.Require().NoError(s.StopController(s.ControllerNode(0)))
+	})
+
+	s.T().Run("Only storage extension enabled", func(t *testing.T) {
+		s.PutFile(s.ControllerNode(0), "/tmp/k0s.yaml", k0sConfigWithStorage)
+		s.Require().NoError(s.InitController(0, "--config=/tmp/k0s.yaml", "--disable-components=konnectivity-server,metrics-server"))
+
+		kc, err := s.KubeClient("controller0", "")
+		s.Require().NoError(err)
+
+		err = s.WaitForNodeReady("worker0", kc)
+		s.NoError(err)
+
+		// When both storage and helm are enabled, there should be no action.
+		// Unfortunately to check that, the best we can do is seeing that it's not
+		// created after a grace period
+		s.T().Log("Checking that the chart is created")
+		s.Require().NoError(s.waitForChartUpdated(ctx, "3.3.0"))
+
+		s.T().Log("Stopping k0s")
+		s.Require().NoError(s.StopController(s.ControllerNode(0)))
+	})
+
+	s.T().Run("Only helm extension enabled", func(t *testing.T) {
+		s.PutFile(s.ControllerNode(0), "/tmp/k0s.yaml", k0sConfigWithHelm)
+		s.Require().NoError(s.InitController(0, "--config=/tmp/k0s.yaml", "--disable-components=konnectivity-server,metrics-server"))
+
+		kc, err := s.KubeClient("controller0", "")
+		s.Require().NoError(err)
+
+		err = s.WaitForNodeReady("worker0", kc)
+		s.NoError(err)
+
+		// When both storage and helm are enabled, there should be no action.
+		// Unfortunately to check that, the best we can do is seeing that it's not
+		// created after a grace period
+		s.T().Log("Checking that the chart is updated to 3.9.0")
+		s.Require().NoError(s.waitForChartUpdated(ctx, "3.9.0"))
+	})
+}
+
+func TestOpenEBSSuite(t *testing.T) {
+	s := OpenEBSSuite{
+		common.BootlooseSuite{
+			ControllerCount: 1,
+			WorkerCount:     2,
```

Comment:
Any special reason why we use two workers here?

Reply:
Thanks for catching that, that's something that I kept by accident from previous work.